mlir/test/mlir-miopen-driver/populate_host_print.mlir

```diff
@@ -0,0 +1,119 @@
+// RUN: mlir-miopen-driver -p -ph -pr | FileCheck %s --check-prefix=F32
+// RUN: mlir-miopen-driver -p -ph -pr -t f16 | FileCheck %s --check-prefix=F16
+// RUN: mlir-miopen-driver -p -ph -pr -t bf16 | FileCheck %s --check-prefix=BF16
+
+// F32-LABEL: func @main()
+// F32-NEXT: alloc() : memref<128x8x3x3xf32>
+// F32-NEXT: alloc() : memref<128x8x32x32xf32>
+// F32-NEXT: alloc() : memref<128x128x30x30xf32>
+// F32-NEXT: alloc() : memref<128x128x30x30xf32>
+// F32-NEXT: memref_cast {{.*}} : memref<128x8x3x3xf32> to memref<?x?x?x?xf32>
+// F32-NEXT: memref_cast {{.*}} : memref<128x8x32x32xf32> to memref<?x?x?x?xf32>
+// F32-NEXT: memref_cast {{.*}} : memref<128x128x30x30xf32> to memref<?x?x?x?xf32>
+// F32-NEXT: constant 1.000000e+00 : f32
+// F32-NEXT: constant 0.000000e+00 : f32
+// F32-NEXT: call @mcpuMemset4DFloat({{.*}}, {{.*}}) : (memref<?x?x?x?xf32>, f32) -> ()
+// F32-NEXT: call @mcpuMemset4DFloat({{.*}}, {{.*}}) : (memref<?x?x?x?xf32>, f32) -> ()
+// F32-NEXT: call @mcpuMemset4DFloat({{.*}}, {{.*}}) : (memref<?x?x?x?xf32>, f32) -> ()
+// F32-NEXT: call @gpu_conv({{.*}}, {{.*}}, {{.*}}) : (memref<128x8x3x3xf32>, memref<128x8x32x32xf32>, memref<128x128x30x30xf32>) -> ()
+// F32-NEXT: constant 0 : index
+// F32-NEXT: constant 1 : index
+// F32-NEXT: constant 128 : index
+// F32-NEXT: constant 128 : index
+// F32-NEXT: constant 30 : index
+// F32-NEXT: constant 30 : index
+// F32-NEXT: scf.for %{{.*}} = %{{.*}} to %{{.*}} step %{{.*}} {
+// F32-NEXT:   scf.for %{{.*}} = %{{.*}} to %{{.*}} step %{{.*}} {
+// F32-NEXT:     scf.for %{{.*}} = %{{.*}} to %{{.*}} step %{{.*}} {
+// F32-NEXT:       scf.for %{{.*}} = %{{.*}} to %{{.*}} step %{{.*}} {
+// F32-NEXT:         %{{.*}} = load %{{.*}}[%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}] : memref<128x128x30x30xf32>
+// F32-NEXT:         store %{{.*}}, %{{.*}}[%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}] : memref<128x128x30x30xf32>
+// F32-NEXT:       }
+// F32-NEXT:     }
+// F32-NEXT:   }
+// F32-NEXT: }
+// F32-NEXT: memref_cast %{{.*}} : memref<128x128x30x30xf32> to memref<*xf32>
+// F32-NEXT: call @print_memref_f32(%{{.*}}) : (memref<*xf32>) -> ()
```

Comment:
Could we populate this into an independent function, like `conv2d` or `gpu_conv` for better readability of IR? This is still in top level of PR and we can manually update those when necessary.

Reply:
will submit a new PR where the print out logic is in a separate function.